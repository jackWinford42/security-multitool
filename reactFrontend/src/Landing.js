import React from "react";
import {useSelector} from "react-redux";

/** Render the landing page
 */
export default function Landing() {
	const user = useSelector(st => st.currUser);
	console.log(user)
	const greeting = (<h4>Welcome back {user.username}</h4>);
	return (
		<div className="Landing">
			<h1>RAMT</h1>
			<h5>home of the Reputation Analyses Multi Tool</h5>
			{(user.username) ? 
			greeting :
			<p id="authButtons">
				<a className="btn btn-primary font-weight-bold mr-3" href="/login">Log in</a>
				<a className="btn btn-primary font-weight-bold" href="/sign-up">Sign up</a>
			</p>}
		</div>
	);
}