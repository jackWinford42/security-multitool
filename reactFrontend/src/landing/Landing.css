#authButtons {
    width: 250px;
    margin: 0 auto;
}

.authButtonsLeft {
    float: left;
}

.authButtonsRight {
    float: right;
}

.Landing {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}