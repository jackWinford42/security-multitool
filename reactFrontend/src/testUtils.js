import React from "react";
import { Provider, useDispatch } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducers/rootReducer";
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from "redux-thunk";

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(thunk)));

const demoUser = {
  email: "jackwinford@gmail.com",
  username: "captainjack",
  password: "qwerty",
  photo_url: "https://bit.ly/3C2VUgQ",
};

// const dispatch = useDispatch();
// await dispatch({type: "BEGIN_AUTH_SESSION", user: demoUser});

const UserProvider =(
    ({children}) =>
    (<Provider store={store}>
      {children}
    </Provider>)
);

export { UserProvider };