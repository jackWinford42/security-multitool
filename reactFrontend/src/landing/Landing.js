import React from "react";
import Button from "@material-ui/core/Button";
import Link from '@material-ui/core/Link';
import { Link as RouterLink } from 'react-router-dom';
import "./Landing.css";

/** Render the landing page
 */
export default function Landing() {
	return (
		<div className="Landing">
			<h1>RAMT</h1>
			<h5>home of the Reputation Analyses Multi Tool</h5>
			<p id="authButtons">
        <Link component={RouterLink} to="/sign-up" className="authButtonsLeft">
          <Button variant="contained">sign up</Button>
        </Link>
        <Link component={RouterLink} to="/login" className="authButtonsRight">
          <Button variant="contained">log in</Button>
        </Link>
			</p>
		</div>
	);
}