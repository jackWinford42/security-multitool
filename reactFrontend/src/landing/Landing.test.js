import React from "react";
import { render } from "@testing-library/react";
import Landing from "./Landing";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils"

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Landing/>
      </UserProvider>
    </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Landing/>
      </UserProvider>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});