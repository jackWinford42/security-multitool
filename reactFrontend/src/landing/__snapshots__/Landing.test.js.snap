// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Landing"
  >
    <h1>
      RAMT
    </h1>
    <h5>
      home of the Reputation Analyses Multi Tool
    </h5>
    <p
      id="authButtons"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover authButtonsLeft MuiTypography-colorPrimary"
        href="/sign-up"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            sign up
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover authButtonsRight MuiTypography-colorPrimary"
        href="/login"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            log in
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
    </p>
  </div>
</DocumentFragment>
`;
